baseservice/iodevices/base/common.py

```diff
@@ -49,9 +65,7 @@ def __eq__(self, other):
         return self.bytes == other.bytes and self._headers == other._headers
 
     def __copy__(self):
-        stream_copy = copy.copy(self._stream)
-        stream_copy.seek(0)
-        return Message(stream_copy, self._headers.copy())
+        return self.copy()
 
     def __deepcopy__(self, memo=None):
```

Comment:
I'm not sure...

Reply:
Are you sure it will be the expected behavior? Regardless, you might want to implement this logic using either __getstate__ or __setstate__ instead. Deepcopy adheres to the pickle protocol, you will maintain this behavior for every other python feature that uses the protocol. It will also simplify the implementation as you won't need to explicitly work with the memo.